### PR TITLE
feat(mysqlnode-otel): refine MYSQLNODE OTel synthesis tags/log rule and add OTel-app relationship (staging)

### PR DESCRIPTION
**Pull request by abhinav1602**
### Relevant information

Follow-up to #3034, refining the staging OpenTelemetry support for the
`INFRA-MYSQLNODE` entity. All changes are **staging-only** (`.stg`); production
behavior is unchanged. Net diff is 3 files.

**What changed**

- **`entity-types/infra-mysqlnode/definition.stg.yml`**
  - Added the `otel.library.version → instrumentation.version` tag on the OTel rules
    (parity with the Postgres/MSSQL OTel definitions).
  - Exposed `server.address` as `entityTagNames: [server.address, endpoint]` and made
    `server.port` explicit — these are the tags the MYSQLNODE relationship candidate
    keys on.
  - Collapsed the two hardcoded `otel.library.name` log rules (contrib + newrelic-forks)
    into a single **receiver-agnostic** rule keyed on `db.system.name: mysql`. Both
    receivers emit `db.system.name=mysql`, so coverage is unchanged and no longer depends
    on hardcoded library paths.
  - Dropped the `db.namespace → database.name` tag from the log rule: it is per-query
    (not instance-scoped), frequently empty, and absent on `db.server.top_query`; the
    query-performance dashboard reads `db.namespace` directly off the log events, so this
    has no functional impact.

- **`relationships/synthesis/OTEL-APPLICATION-to-INFRA-MYSQLNODE.stg.yml`** *(new)*
  - `CALLS` relationship from OTel-instrumented applications to the MySQL node, resolved
    from OTLP distributed-tracing spans (`newrelic.source = api.traces.otlp`) via the
    `MYSQLNODE` candidate. Two rules cover both `db.system` and `db.system.name` semantic
    conventions. Complements the existing native-APM relationship
    (`APM-APPLICATION-to-INFRA-MYSQLNODE.stg.yml`) and mirrors
    `OTEL-APPLICATION-to-INFRA-MSSQLINSTANCE.yml`.

- **`relationships/candidates/MYSQLNODE.stg.yml`**
  - Comment-only: noted the candidate is now consumed by both the native-APM and OTLP
    synthesis rules.

**Note / prerequisite**

The MYSQLNODE candidate uses `matchingMode: ALL`, so both `server.address` and
`server.port` tags must be present on the entity for app→MySQL relationships to resolve.
The `mysqlreceiver` emits only `mysql.instance.endpoint`, so the collector must derive
`server.address`/`server.port` (e.g. a `transform`/`resource` processor). To verify in
staging before promotion.

**Validation**

`npm --prefix validator run check` passes (schemas + synthesis rules + relationship rules).

### Checklist

* [x] I've read the guidelines and understand the acceptance criteria.
* [x] The value of the attribute marked as `identifier` will be unique and valid.
* [x] I've confirmed that my entity type wasn't already defined.

### CODE REVIEW

**Review by entityBot — DISMISSED** [review]
We don't think these changes require an API Review Board (ARB) review, so it is not required to merge this PR. Note that all API changes do require one: if you believe this PR needs an ARB review, please create the ticket and add the `ARB:pending` label. Reach `#api-review-board` in case of doubts.

**Review by spathlavath — DISMISSED** [review]

File: entity-types/infra-mysqlnode/definition.stg.yml [hunk 1]
```diff
@@ -23,62 +23,43 @@ synthesis:
...
       encodeIdentifierInGUID: true
       conditions:
         - attribute: eventType
           value: Log
-        - attribute: event.name
-          prefix: db.server.
         - attribute: instrumentation.provider
           value: opentelemetry
-        - attribute: otel.library.name
-          value: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mysqlreceiver
```
Comment (mraju1) [inline entity-types/infra-mysqlnode/definition.stg.yml:48]: 👍

File: relationships/synthesis/OTEL-APPLICATION-to-INFRA-MYSQLNODE.stg.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          fields:
+            - field: serverAddress
+              attribute: server.address
+            - field: serverPort
+              attribute: server.port
+
+  - name: otelApplicationCallsMysqlNodeFromTracesSemconv
+    version: "1"
+    origins:
+      - Distributed Tracing
```
Comment (mraju1) [inline relationships/synthesis/OTEL-APPLICATION-to-INFRA-MYSQLNODE.stg.yml:42]: curious why postgres looks for APM metrics and mysql one looks for distrbuted tracing

**Review by mraju1 — DISMISSED** [review]

File: relationships/synthesis/OTEL-APPLICATION-to-INFRA-MYSQLNODE.stg.yml (hunk 2 above)
Comment (abhinav1602) [reply relationships/synthesis/OTEL-APPLICATION-to-INFRA-MYSQLNODE.stg.yml:42]: When an APM agent instruments an application making a database call, it emits a datastore span.

By setting origins: Distributed Tracing, we’re telling the system to rely strictly on span/trace data when evaluating this relationship. Since distributed tracing is [enabled by default with APM](https://docs.newrelic.com/docs/apm/agents/java-agent/configuration/distributed-tracing-java-agent/#quick-start-apm), this feels like a safe default for now.

That said, we’ll get a clearer picture of any edge cases once we merge and test this in staging. If we run into any issues, we can easily roll back to the non-DT approach. 

Let me know what you think!

**Review by entityBot — APPROVED** [review]
We don't think these changes require an API Review Board (ARB) review, so it is not required to merge this PR. Note that all API changes do require one: if you believe this PR needs an ARB review, please create the ticket and add the `ARB:pending` label. Reach `#api-review-board` in case of doubts.

File: relationships/synthesis/OTEL-APPLICATION-to-INFRA-MYSQLNODE.stg.yml (hunk 2 above)
Comment (abhinav1602) [reply relationships/synthesis/OTEL-APPLICATION-to-INFRA-MYSQLNODE.stg.yml:42]: @mraju1 Just to be on the safe side, I added another rule to cover metrics as well.

**Review by spathlavath — APPROVED** [review]

**Review by mborroni — APPROVED** [review]
